hummingbot/strategy/arbitrage/arbitrage.pyx

```diff
@@ -372,6 +344,7 @@ cdef class ArbitrageStrategy(StrategyBase):
             MarketBase buy_market = buy_market_trading_pair_tuple.market
             MarketBase sell_market = sell_market_trading_pair_tuple.market
 
+        buy_price, sell_price = self.c_get_arbitrage_pair_prices(buy_1_sell_2)
```

Comment:
The `buy_price` and `sell_price` should be the price at which the bot can buy/sell with the `best_amount`, not the top bid or top ask. 
You can modify the `c_find_best_profitable_amount` to also return price (not just amount), also can rename the function accordingly.

Reply:
Done!